src/mlpack/tests/decision_tree_test.cpp

```diff
@@ -152,12 +306,30 @@ BOOST_AUTO_TEST_CASE(InformationGainEvenSplitManyClassTest)
       labels[i] = i;
 
     // Calculate information gain and make sure it is correct.
-    BOOST_REQUIRE_CLOSE(InformationGain::Evaluate(labels, c),
+    BOOST_REQUIRE_CLOSE(InformationGain::Evaluate<false>(labels, c, weights),
         std::log2(1.0 / c), 1e-5);
   }
 }
 
 /**
+ * Test the information gain with weighted labels
+ */
+BOOST_AUTO_TEST_CASE(InformationWithWeight)
+{
+  arma::Row<size_t> labels(10);
+  arma::rowvec weights("1 1 1 1 1 0 0 0 0 0");
+  for (size_t i = 0; i < 5; ++i)
+    labels[i] = 0;
+  for (size_t i = 5; i < 10; ++i)
+    labels[i] = 1;
+
+  // Zero is not a good result as gain, but we just need to prove cacluation works.
+  BOOST_REQUIRE_CLOSE(InformationGain::Evaluate<true>(labels, 2, weights), 0, 1e-5);
```

Comment:
Zero is the best possible result for information gain, so I don't think this is 'not a good result'. :)